### ISSUE TITLE
Presto can't query the Hudi Table with metadata mode is dfs

### ISSUE THREAD

**Issue by cx707662950** [prestodb/presto#19083]
example:

CREATE CATALOG hudi_dfs_catalog
    WITH (
        'type'='hudi',
        'catalog.path' = 'hdfs://nn:8020/hudi_dfs_catalog',
        'mode'='dfs'
    );

**Comment by pratyakshsharma** [comment]
As per the documentation, there is no such command as `CREATE CATALOG` in presto. Do you mind sharing the output of this sample statement you shared?

**Comment by cx707662950** [comment]
> As per the documentation, there is no such command as `CREATE CATALOG` in presto. Do you mind sharing the output of this sample statement you shared?

thanks for your reply, actually, I use flink sql to create CATALOG ,then, Presto to  read data, I mean presto can't read catalog type when mode='dfs'

**Comment by pratyakshsharma** [comment]
Presto needs pre configured catalogs for querying the underlying data source. Do you have hudi.properties file in your etc/catalog folder? Presto will not be able to read catalog otherwise.

**Comment by cx707662950** [comment]
> Presto needs pre configured catalogs for querying the underlying data source. Do you have hudi.properties file in your etc/catalog folder? Presto will not be able to read catalog otherwise.
yes, the hudi.properties file in etc/catalog, I mean, presto on hudi only support hive metastore to store metadata, but can't support store metadata in dfs(like hdfs)

**Comment by pratyakshsharma** [comment]
Do you mind sharing your hudi.properties file?

**Comment by cx707662950** [comment]
> Do you mind sharing your hudi.properties file?
Thanks for your help,
below is the  mode is HMS:
----------------------------------------
connector.name=hudi
hive.metastore.uri=thrift://node1:9083

but i don't know how to set properties when mode is DFS

**Comment by pratyakshsharma** [comment]
Right, please check this class https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastoreConfig.java. 

Essentially you need to set the configs mentioned there. Hope this helps.

**Comment by cx707662950** [comment]
> Right, please check this class https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastoreConfig.java.
> 
> Essentially you need to set the configs mentioned there. Hope this helps.

thanks very much.
